app/scripts/metamask-controller.js

```diff
@@ -127,6 +131,14 @@ module.exports = class MetamaskController extends EventEmitter {
       provider: this.provider,
       blockTracker: this.blockTracker,
     })
+    // start and stop polling for balances based on activeControllerConnections
+    this.on('controllerConnectionChanged', (activeControllerConnections) => {
```

Comment:
These changes to the metamask-controller seem to be duplicating functionality we already have in another method. We should maybe add the start/stop calls to `isClientOpen`:

https://github.com/MetaMask/metamask-extension/blob/a2654108bed88db5cd09c22049471c5c3f1199d6/app/scripts/metamask-controller.js#L1452-L1467